---
title: "Class 19: Investigating pertussis resurgence Mini Project"
format: pdf
---

# Web Scrpping

Here I extract the CDC figures for pertussis in the US
https://www.cdc.gov/pertussis/surv-reporting/cases-by-year.html

```{r echo = FALSE}
cdc <- data.frame(
                                 Year = c(1922L,1923L,1924L,1925L,
                                          1926L,1927L,1928L,1929L,1930L,1931L,
                                          1932L,1933L,1934L,1935L,1936L,
                                          1937L,1938L,1939L,1940L,1941L,1942L,
                                          1943L,1944L,1945L,1946L,1947L,
                                          1948L,1949L,1950L,1951L,1952L,
                                          1953L,1954L,1955L,1956L,1957L,1958L,
                                          1959L,1960L,1961L,1962L,1963L,
                                          1964L,1965L,1966L,1967L,1968L,1969L,
                                          1970L,1971L,1972L,1973L,1974L,
                                          1975L,1976L,1977L,1978L,1979L,1980L,
                                          1981L,1982L,1983L,1984L,1985L,
                                          1986L,1987L,1988L,1989L,1990L,
                                          1991L,1992L,1993L,1994L,1995L,1996L,
                                          1997L,1998L,1999L,2000L,2001L,
                                          2002L,2003L,2004L,2005L,2006L,2007L,
                                          2008L,2009L,2010L,2011L,2012L,
                                          2013L,2014L,2015L,2016L,2017L,2018L,
                                          2019L),
         Cases = c(107473,164191,165418,152003,
                                          202210,181411,161799,197371,
                                          166914,172559,215343,179135,265269,
                                          180518,147237,214652,227319,103188,
                                          183866,222202,191383,191890,109873,
                                          133792,109860,156517,74715,69479,
                                          120718,68687,45030,37129,60886,
                                          62786,31732,28295,32148,40005,
                                          14809,11468,17749,17135,13005,6799,
                                          7717,9718,4810,3285,4249,3036,
                                          3287,1759,2402,1738,1010,2177,2063,
                                          1623,1730,1248,1895,2463,2276,
                                          3589,4195,2823,3450,4157,4570,
                                          2719,4083,6586,4617,5137,7796,6564,
                                          7405,7298,7867,7580,9771,11647,
                                          25827,25616,15632,10454,13278,
                                          16858,27550,18719,48277,28639,32971,
                                          20762,17972,18975,15609,18617)
       )
```

lets make a plot of the number of cases of pertussis per year

```{r}
library(ggplot2)
```
```{r}
base <- ggplot(cdc) +
  aes(Year, Cases) +
  geom_point() +
  geom_line() +
  scale_y_continuous(labels = scales::label_comma())

base
```
the first vaccine (WP) was introduced in 1946, lets add this as a line to the plot


```{r}
base + geom_vline(xintercept = 1946, col = "blue")
```
in 1996 the switch to the AP vaccine occurred, lets add this to our plot as a red line

```{r}
base + geom_vline(xintercept = 1946, col = "blue") + geom_vline(xintercept = 1996, col = "red" )
```

# Exploring CBI-PB data

Why is this vaccine-preventable disease on the upswing? To answer this question we need to investigate the mechanisms underlying waning protection against pertussis. This requires evaluation of pertussis-specific immune responses over time in wP and aP vaccinated individuals.

The new and ongoing CMI-PB project aims to provide the scientific community with this very information.

The CMI-PB API (like most APIs) sends responses in JSON format. We will use **jsonlite**

```{r}
library(jsonlite)
```

```{r}
subject <- read_json("http://cmi-pb.org/api/subject", simplifyVector = TRUE)

head(subject)

dim(subject)
```

> Q. How many wP and aP subjects are there?

```{r}
table(subject$infancy_vac)
```


> Q. How many female, non-white individuals are there?

```{r}
table(subject$race, subject$biological_sex)
```

lets look at the specimen data

```{r}
specimen <- read_json("http://cmi-pb.org/api/specimen", simplifyVector = TRUE)

head(specimen)

dim(specimen)
```

To know whether a given specimen_id comes from an aP or wP individual we need to link (a.k.a. “join” or merge) our specimen and subject data frames. The excellent dplyr package (that we have used previously) has a family of join() functions that can help us with this common task:

```{r}
library(dplyr)

meta <- inner_join(specimen, subject)

dim(meta)

head(meta)
```

titer data

```{r}
titer <- read_json("http://cmi-pb.org/api/ab_titer", simplifyVector = TRUE)

dim(titer)

head(titer)
```

> Q. How many isotypes are there in this dataset?

```{r}
table(titer$isotype)
```

join the titer data with meta

```{r}
abdata <- inner_join(titer, meta)

dim(abdata)

head(abdata)
```

> Q. What do you notice about the "visit" number 8?

```{r}
table(abdata$visit)
```

the project is ongoing so we will just look at visits 1-7

# Looking at IgG1 ab levels

```{r}
ig1 <- filter(abdata, isotype == "IgG1", visit!= 8)

dim(ig1)

head(ig1)

```

> Q. How many different antigens are there?

```{r}
table(abdata$antigen)
```


Analysis of the whole data set: antigen levels i.e a plot of antigen vs MFI

```{r}
plot <- ggplot(ig1) +
  aes(MFI, antigen) +
  geom_boxplot()


plot
```
adding faceting by visit

```{r}
visit_plot <- plot + facet_wrap(vars(visit), nrow=2)

visit_plot
```

> Q. Is there a difference btwn wP and aP individuals?

```{r}
visit_plot + aes(MFI, antigen, col = infancy_vac)
```

```{r}
ggplot(ig1) +
  aes(MFI, antigen, col=infancy_vac ) +
  geom_boxplot(show.legend = FALSE) + 
  facet_wrap(vars(infancy_vac, visit), nrow=2)+
  theme(axis.text.x = element_text(angle = 45, hjust=1))

```

FIM2/3 antigen levels 

```{r}
fim23 <- filter(ig1, antigen == "FIM2/3")

ggplot(fim23) +
  aes(MFI, col = infancy_vac) +
  geom_boxplot() +
  facet_wrap(vars(visit))
```

PT

```{r}
pt <- filter(ig1, antigen == "PT")

ggplot(pt) +
  aes(MFI, col = infancy_vac) +
  geom_boxplot() +
  facet_wrap(vars(visit))
```

> Q. Is there a diff btwn aP and wP?

No, there isnt a clear diff btwn ab response levels btwn aP and wP

# RNASeq data

For RNA-Seq data the API query mechanism quickly hits the web browser interface limit for file size. We will present alternative download mechanisms for larger CMI-PB datasets in the next section. However, we can still do “targeted” RNA-Seq querys via the web accessible API.

For example we can obtain RNA-Seq results for a specific ENSEMBLE gene identifier or multiple identifiers combined with the & character


```{r}
url <- "https://www.cmi-pb.org/api/v2/rnaseq?versioned_ensembl_gene_id=eq.ENSG00000211896.7"

rna <- read_json(url, simplifyVector = TRUE)
```


join with meta data to get all the important info in on e spot

```{r}
ssrna <- inner_join(rna, meta)

dim(ssrna)

head(ssrna)
```

> Q. plot of timecourse of gene expression for IgG1


```{r}
ggplot(ssrna) +
  aes(visit, tpm, group= subject_id) +
  geom_point() +
  geom_line(alpha = 0.2)
```
> Q19. What do you notice about the expression of this gene (i.e. when is it at it’s maximum level)?

peaks at visit 4

> Q20. Does this pattern in time match the trend of antibody titer data? If not, why not?

It predates the antibody titer data, which is max at visit 5
